src/transformers/models/t5/modeling_t5.py

```diff
@@ -91,14 +92,17 @@ def rotate_half(x):
 
 
 #@torch.jit.script
-def apply_rotary_pos_emb(x, cos, sin):
+def apply_rotary_pos_emb(x, cos, sin, seq_dimension):
     # NOTE: This could probably be moved to Triton
 
-    # Handle a possible sequence length mismatch in between q and k
-    cos = cos[:, :, : x.shape[-2], :]
-    sin = sin[:, :, : x.shape[-2], :]
+    # # Handle a possible sequence length mismatch in between q and k
+    # cos = cos[:, :, : x.shape[-2], :]
+    # sin = sin[:, :, : x.shape[-2], :]
+    res = torch.clone(x)
```

Comment:
why the clone here?

Reply:
It's probably not required - but since x is cached, I did want to touch it to be on the safe side.